openedx/openedx-event-sink-clickhouse
# Remove block relationships / add email to profile sink

**Description:** 

*Breaking changes*
1- Removes the xblock relationships since they are expensive, unused, and difficult to work with. We should use the section/subsection/unit included on the blocks themselves.
2- Add the email address of the user to the user profile sink


**Testing instructions:**
1- You will need to add a column to your ClickHouse user profile table         
```
ALTER TABLE {{ ASPECTS_EVENT_SINK_DATABASE }}.{{ ASPECTS_EVENT_SINK_USER_PROFILE_TABLE }} 
        ADD COLUMN IF NOT EXISTS email String DEFAULT '' 
        AFTER name
``` 
2- Use this branch for your openedx Tutor containers, I usually just change `OPENEDX_EXTRA_PIP_REQUIREMENTS` to override the current version with `git+https://github.com/openedx/openedx-event-sink-clickhouse.git@bmtcril/remove_block_relationships`

3- `tutor images build openedx --no-cache`
4- Start tutor 
5- Test the email change by updating anything in your profile. You should see a new row in the user profile that has the correct email address.
5- Test the relationships by publishing a course. You should not see any new relationships rows in `event_sink.course_relationships`, but should see new dumps for `course_blocks` and `course_overviews`. Just check the `time_last_dumped` for each.

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** Updates will need to be made to Aspects Alembic migrations before it can be updated with this.
